packages/shared/notionable-hooks/src/market/use-market.ts

```diff
@@ -17,21 +17,54 @@ export const useAllMarkets = () => {
 
   const orderedCurrencyIds = Array.from(currencyMarkets.keys()).sort();
 
-  const rates = orderedCurrencyIds.map((i) => {
+  let allRates = {}
+  orderedCurrencyIds.map((i) => {
     try {
       const { orderedMarkets } = currencyMarkets.get(i)!;
-
-      return orderedMarkets.reduce((maxRate: number, m) => {
-        const rate = m.marketAnnualizedRate();
-        if (maxRate === 0) return rate;
-
-        return rate > maxRate ? rate : maxRate;
-      }, 0);
+      return orderedMarkets.map((data, index) => {
+        if(!allRates[data.underlyingSymbol] && data.marketAnnualizedRate() > 0){
+          allRates = {
+            [data.underlyingSymbol]: {
+              [index]: data.marketAnnualizedRate(),
+            },
+            ...allRates    
+          }
+        }
+        if(data.marketAnnualizedRate() > 0){
+          allRates[data.underlyingSymbol] = {
+            [index]: data.marketAnnualizedRate(),
+            ...allRates[data.underlyingSymbol]
+          }
+        }
+        // NOTE* this return is just to please the linter
```

Comment:
If you're going to use an array here, you should be able to return from map directly and then you can remove the // NOTE comment